packages/core/tests/test_precompact_dump.py

```
@@ -0,0 +1,163 @@
+        assert Path(path).parent == shared_dir
+        assert Path(path).name.startswith("precompact-")
+        assert Path(path).name.endswith(".md")
+        assert Path(path).read_text(encoding="utf-8") == "# hello\n"
+
+    def test_prunes_old_dumps(self, tmp_path: Path) -> None:
+        """古いダンプは MAX_DUMP_FILES を超えると削除され、新規 dump は残ることを確認する。"""
+        project_dir = str(tmp_path)
+        shared_dir = tmp_path / ".claude" / "context" / "shared"
+        shared_dir.mkdir(parents=True)
+
+        # 既存ダンプを 21 個作る（閾値 +1）。
```

Ruff (0.15.9)

[warning] Comment contains ambiguous `（` (FULLWIDTH LEFT PARENTHESIS). Did you mean `(` (LEFT PARENTHESIS)?

(RUF003)